tls/s2n_psk.h

```diff
@@ -98,11 +98,16 @@ int s2n_psk_set_identity(struct s2n_psk *psk, const uint8_t *identity, uint16_t
 int s2n_psk_set_secret(struct s2n_psk *psk, const uint8_t *secret, uint16_t secret_size);
 int s2n_psk_set_hmac(struct s2n_psk *psk, s2n_psk_hmac hmac);
 
+int s2n_psk_get_type(struct s2n_psk *psk, uint8_t *type);
```

Comment:
Type needs to be an enum, not a uint8_t. See `s2n_connection_set_psk_mode`. 

Additionally, with the introduction of a more global "mode", I'm not sure this API should exist. Do we actually need it for integration tests? The case where the connection is acting like it negotiated an external PSK but has that PSK marked as a resumption PSK seems like a matter for unit tests.